src/libsyntax/visit.rs

```diff
@@ -549,12 +550,8 @@ pub fn walk_fn_ret_ty<'a, V: Visitor<'a>>(visitor: &mut V, ret_ty: &'a FunctionR
 }
 
 pub fn walk_fn_decl<'a, V: Visitor<'a>>(visitor: &mut V, function_declaration: &'a FnDecl) {
-    for argument in &function_declaration.inputs {
-        walk_list!(visitor, visit_attribute, argument.attrs.iter());
-        visitor.visit_pat(&argument.pat);
-        visitor.visit_ty(&argument.ty);
-    }
-    visitor.visit_fn_ret_ty(&function_declaration.output)
+    function_declaration.inputs.iter().for_each(|arg| visitor.visit_arg(arg));
```

Comment:
I don't like using for_each on containers too much. It should mainly be used for long iterator chains

Reply:
Nit: `for_each` because it cleaner to have one line instead of three and this is not a "breakable" loop